### DISCUSSION TITLE
x-for Components.

### DISCUSSION THREAD

**Discussion by valluminarias** [alpinejs/alpine#792] [4. General]
Hi.
I wanted to loop an array into components. 
What I want to achieve is like this:
```
<template x-for="(el, index) in arr">

    <div x-data="{}">
        // Component Data
        ...
          How can I retrieve the `el` variable in the loop?
        ...
    </div>

</template>
```

Please help.
Thanks.

**Comment by jasonrockacy-jobvite** [comment]
Hi @valluminarias ,

You need to surround your `<template>` tags with your `x-data={}` element. See below:
```
<div x-data="{arr:[{obj1}, {obj2}, ...]}"> // x-data should be your top-level element
        // Component Data
       <template x-for="(obj, index) in arr"> // Now alpine knows what to do with the template tag
              // Array data
       </template>
</div>
```

The `x-data` attribute is what initializes an instance of AlpineJS. Therefore, everything inside the element that contains the `x-data` attribute will be used by AlpineJS, otherwise the template tag is not in scope. Hopefully, this makes sense. :)

Kind Regards,
Jason R.

**Reply by valluminarias** [reply]
Hi @jasonrockacy-jobvite, how about having some `x-data` or component inside the `x-for` loop?

**Reply by HugoDF** [reply]
You can do that but you won't be able to access the data from the loop/parent component
